src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/build/common.targets

```diff
@@ -65,6 +66,12 @@
         <RelativePath>$([System.IO.Path]::GetFileName('$(_MvcRazorOutputFullPath)'))</RelativePath>
       </ResolvedFileToPublish>
     </ItemGroup>
+
+    <ItemGroup Condition="'$(MvcRazorExcludeRefAssembliesFromPublish)'=='true'">
+      <ResolvedFileToPublish
+        Remove="%(ResolvedFileToPublish.Identity)"
+        Condition="'%(ResolvedFileToPublish.RelativePath)'=='$(RefAssembliesFolderName)\%(Filename)%(Extension)'" />
```

Comment:
Using https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.PreserveCompilationContext.targets#L55 as the source of the condition.